examples/src/bin/ota_update.rs

```diff
@@ -0,0 +1,158 @@
+    }
+
+    // Find the OTA-data partition and show the currently active partition
+    let ota_part = pt
+        .find_partition(esp_bootloader_esp_idf::partitions::PartitionType::Data(
+            DataPartitionSubType::Ota,
+        ))
+        .unwrap()
+        .unwrap();
+    let mut ota_part = ota_part.as_embedded_storage(&mut storage);
+    println!("Found ota data");
+
+    let mut ota = esp_bootloader_esp_idf::ota::Ota::new(&mut ota_part).unwrap();
+    let current = ota.current_slot().unwrap();
+    println!(
+        "current image state {:?} (only relevant if the bootloader was built with auto-rollback support)",
+        ota.current_ota_state()
+    );
+    println!("current {:?} - next {:?}", current, current.next());
+
+    // Mark the current slot as VALID - this is only needed if the bootloader was built with auto-rollback support.
+    // The default pre-compiled bootloader in espflash is NOT.
+    if ota.current_slot().unwrap() != Slot::None
+        && (ota.current_ota_state().unwrap() == esp_bootloader_esp_idf::ota::OtaImageState::New
+            || ota.current_ota_state().unwrap()
+                == esp_bootloader_esp_idf::ota::OtaImageState::PendingVerify)
+    {
+        println!("Changed state to VALID");
+        ota.set_current_ota_state(esp_bootloader_esp_idf::ota::OtaImageState::Valid)
+            .unwrap();
+    }
+
+    cfg_if::cfg_if! {
+        if #[cfg(any(feature = "esp32", feature = "esp32s2", feature = "esp32s3"))] {
+            let button = peripherals.GPIO0;
+        } else {
+            let button = peripherals.GPIO9;
+        }
+    }
+
+    let boot_button = Input::new(button, InputConfig::default().with_pull(Pull::Up));
+
+    println!("Press boot button to flash and switch to the next OTA slot");
+    let mut done = false;
+    loop {
+        if boot_button.is_low() && !done {
+            done = true;
+
+            let next_slot = current.next();
+
+            println!("Flashing image to {:?}", next_slot);
+
+            // find the target app partition
+            let next_app_partition = match next_slot {
+                Slot::None => {
+                    // None is FACTORY if present, OTA0 otherwise
+                    pt.find_partition(partitions::PartitionType::App(AppPartitionSubType::Factory))
+                        .or_else(|_| {
+                            pt.find_partition(partitions::PartitionType::App(
+                                AppPartitionSubType::Ota0,
+                            ))
+                        })
+                        .unwrap()
+                }
+                Slot::Slot0 => pt
+                    .find_partition(partitions::PartitionType::App(AppPartitionSubType::Ota0))
+                    .unwrap(),
+                Slot::Slot1 => pt
+                    .find_partition(partitions::PartitionType::App(AppPartitionSubType::Ota1))
+                    .unwrap(),
+            }
+            .unwrap();
```

Comment:
I'm fine with writing this logic by hand if needed, but couldn't we provide some utility function for some common cases, like any setup with 2+ OTA partitions? As a user I'd be happy if I could get away with at most this much (pseudo)code:

```rust
let mut updater = start_update().unwrap(); // Could fail if there is no "next" update slot
for chunk in http_response.read_next() {
   updater.write(chunk).unwrap(); // Could fail data overflows the partition
}
updater.activate().unwrap(); // Could fail a checksum check
```